core/src/Kokkos_Macros.hpp

```diff
@@ -356,8 +356,8 @@
 #endif
 #endif
 
-#if (1700 > KOKKOS_COMPILER_INTEL)
```

Comment:
I guess we could just enable it for `Intel` in general to be on the safe side?

Reply:
I was not able to reproduce the error with Intel 17.  My answer is yes we could but I am not changing unless someone request it.